index.js

```diff
@@ -56,8 +56,12 @@ module.exports = function (request, options) {
 
     function onEnd(err) {
       if(err) reject(err);
-      cleanup();
-      return resolve({fields, files});
+      Promise.all(filePromises)
```

Comment:
That is a big sub-optimization. If you want performance you should rewrite to get rid of the temporary file stored in /tmp.